src/main/java/net/sf/jabref/logic/importer/fetcher/MedlineFetcher.java

```diff
@@ -0,0 +1,202 @@
+            String inLine;
+            while ((inLine = in.readLine()) != null) {
+
+                // get the count
+                Matcher idMatcher = ID_PATTERN.matcher(inLine);
+                if (idMatcher.find()) {
+                    addID(idMatcher.group(1));
+                }
+                Matcher retMaxMatcher = RET_MAX_PATTERN.matcher(inLine);
+                if (retMaxMatcher.find()) {
+                    retmax = Integer.parseInt(retMaxMatcher.group(1));
+                }
+                Matcher retStartMatcher = RET_START_PATTERN.matcher(inLine);
+                if (retStartMatcher.find()) {
+                    retstart = Integer.parseInt(retStartMatcher.group(1));
+                }
+                Matcher countMatcher = COUNT_PATTERN.matcher(inLine);
+                if (doCount && countMatcher.find()) {
+                    count = Integer.parseInt(countMatcher.group(1));
+                    doCount = false;
+                }
+            }
+        } catch (URISyntaxException | MalformedURLException e) { // new URL() failed
+            LOGGER.warn("Bad url", e);
+        } catch (IOException e) { // openConnection() failed
+            LOGGER.warn("Connection failed", e);
+        }
+    }
+
+    @Override
+    public String getName() {
+        return "Medline";
+    }
+
+    @Override
+    public HelpFile getHelpPage() {
+        return HelpFile.FETCHER_MEDLINE;
+    }
+
+    @Override
+    public Optional<BibEntry> performSearchById(String identifier) throws FetcherException {
+        String cleanQuery = identifier.trim().replace(';', ',');
+
+        List<BibEntry> entry = fetchMedline(cleanQuery);
+
+        if (entry.isEmpty()) {
+            LOGGER.warn(Localization.lang("No references found"));
+        } else if (entry.size() > 1) {
+            LOGGER.info("Fetcher " + getName() + " " +
+                    "found more than one result for identifier " + identifier
+                    + ". We will use the first entry.");
+        }
+        return Optional.of(entry.get(0));
+    }
+
+    @Override
+    public List<BibEntry> performSearch(String query) throws FetcherException {
+        final int NUMBER_TO_FETCH = 50;
+        List<BibEntry> entryList = new LinkedList<>();
+
+        if (!query.isEmpty()) {
+            String searchTerm = replaceCommaWithAND(query);
+
+            // get the ids from entrez
+            getIds(searchTerm, 0, 1);
+
+            if (count == 0) {
+                LOGGER.warn(Localization.lang("No references found"));
+                return Collections.emptyList();
+            }
+
+            getIds(searchTerm, 0, NUMBER_TO_FETCH-1);
+
+            List<BibEntry> bibs = fetchMedline(ids);
+            entryList.addAll(bibs);
+
+            return entryList;
+        }
+        throw new FetcherException(Localization.lang("Input error") + ". " + Localization.lang("Please enter a comma separated list of Medline IDs (numbers) or search terms."));
```

Comment:
I wouldn't throw this exception if the query is empty. Just return an empty list in this case (also change the order: empty query -> empty list, otherwise fetch)